src/components/FilterButton/FilterButton.jsx

```diff
@@ -2,20 +2,24 @@ import { useState } from "react";
 
 export const FilterButton = ({filterOrigin, origin}) => {
   const [buttonColor, setButtonColor] = useState('white');
+  const [buttonSelected, setButtonSelected] = useState(false);
+
   const markButtonSelected = () => {
-    buttonColor === "white" ? setButtonColor("lightBlue") : setButtonColor("white");
+    buttonSelected ? setButtonColor("white") : setButtonColor("lightBlue");
   }
+
   return (
     <button key={origin} className="filterButton"
-    key={origin}
-    onClick={() => {
-    filterOrigin({origin}); 
-    markButtonSelected();
-  }} 
-  style={{
-    backgroundColor: buttonColor,
-    borderRadius: '8px',
-    marginBottom: '20px'
-  }}>{origin}</button>
+      onClick={() => {
+        const isButtonSelected = setButtonSelected(!buttonSelected);
+        filterOrigin({origin, isButtonSelected}) 
+      markButtonSelected();
```

Comment:
This seems completely unnecessary.

When you click a button, it has an `:active` state or at least a `:focus` or `:focus-within`. Thus you could manage the color of the button when clicked entirely through CSS and without any JS logic, inline styles, or classes.

Reply:
Is that still true in the use case where I want the user to be able to make multiple selections and have them maintain color state?

Reply:
If you're doing something like that, you should be using checkboxes.

`<input type="checkbox" ....... />`

Then in CSS you can use 
```
input[type='checkbox'] {
   /* unchecked style goes here */
   
   &:checked {
       /*checked style goes here */
   }
}
```

Reply:
This one I couldn't get. I ended up with a label and a checkbox. I hid the checkbox, and made the label look like a button and put the text in it. But I couldn't nest anything inside the checkbox, so could not sort out how, with just CSS, to change the state of the label based on the state of the checkbox 😒

I assume this one is not worth chasing, but I did push the WIP to feature/change-buttons-to-checkbox in case I come up with extra time to keep poking at it.

Reply:
You're definitely on the right track. That's how it's typically done.

If you have a label next to your checkbox, like so:

```
<input type="checkbox" ... />
<label ...> ... </label>
```

Then with CSS you can do this:

```
input[type="checkbox"] {
   + label {
       /* unchecked label style */
   }
   
   &:checked {
       + label {
           /* checked label style */
       }
    }
}
```

https://developer.mozilla.org/en-US/docs/Web/CSS/Adjacent_sibling_combinator

Reply:
Ahhhh sweet victory. That '+ label' syntax did the trick 😁